sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/ArchiveBasedAnalysisInputLocation.java

```diff
@@ -98,14 +98,13 @@ public Optional<JavaSootClassSource> getClassSource(@Nonnull ClassType type, @No
   @Override
   @Nonnull
   public Collection<JavaSootClassSource> getClassSources(@Nonnull View view) {
-    // we don't use the filesystem cache here as it could close the filesystem after the timeout
-    // while we are still iterating
-    try (FileSystem fs = FileSystems.newFileSystem(path, (ClassLoader) null)) {
+    try {
+      FileSystem fs = fileSystemCache.get(path);
```

Comment:
why not keep it in the try-with-resources statement?

Reply:
filling/using the cache with living entries